packages/collector/src/announceCycle/agentready.js

```diff
@@ -110,14 +124,19 @@ function enter(_ctx) {
 }
 
 function leave() {
-  uncaught.deactivate();
-  metrics.deactivate();
-  transmissionCycle.deactivate();
+  if (isMainThread) {
+    uncaught.deactivate();
+    metrics.deactivate();
+    requestHandler.deactivate();
+    transmissionCycle.deactivate();
+    deScheduleTracingMetrics();
+  }
+
   tracing.deactivate();
-  requestHandler.deactivate();
-  if (tracingMetricsTimeout) {
-    clearTimeout(tracingMetricsTimeout);
-    tracingMetricsTimeout = null;
+
+  if (profiler) {
+    profiler.destroy();
+    profiler = null;
```

Comment:
Destroying the profiler in `agentready#leave` is not directly related to the main intent of this PR, but I noticed that this was missing, so I added it.

Reply:
My impression is that `agentready#leave` is only executed when `ctx.transitionTo('agentready');` is called, but profile will be null then, because this state was never entered. My impression is that `agentready` state can only happen once. (Autoprofile will autodestroy on process exit, which I thought is enough based on the logic. )
Maybe better to discuss offline :)

Reply:
agentready#leave can also be triggered when we lose the connection to the agent (for example, if the agent is stopped or crashes), see https://github.com/instana/nodejs/blob/main/packages/collector/src/announceCycle/agentready.js#L87

When that has happened, we go back to state unannounced and start the announce loop again. If the agent comes back up, we would actually execute agentready#enter a second time.

In that scenario, without an explicit call to profiler.destroy() the profiler from the first successful announce would still be active (although we no longer have a referenced to it from @instana/collector), and we would create and start a second profiler.

Reply:
Ah yeah nice. That makes sense. Thanks <3